src/core/lib/promise/activity.cc

```diff
@@ -0,0 +1,112 @@
+// Copyright 2021 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <grpc/impl/codegen/port_platform.h>
+
+#include "src/core/lib/gprpp/atomic_utils.h"
+#include "src/core/lib/promise/activity.h"
+
+namespace grpc_core {
+
+///////////////////////////////////////////////////////////////////////////////
+// GLOBALS
+
+thread_local Activity* Activity::g_current_activity_ = nullptr;
```

Comment:
is ABSL_CONST_INIT compatible with GPR_THREAD_LOCAL with pthreads?

Reply:
Good question. I can research a bit more. Did you mean to use `GPR_THREAD_LOCAL` here instead of `thread_local`?

Reply:
Ah, ok, you've since changed it to `GPR_THREAD_LOCAL`. I'm not sure why it wouldn't work with `grpc_core::PthreadTlsImpl`, the `[[clang::require_constant_initialization]]` attribute check works with variables of static or thread storage duration. We can run an experiment later, I don't want to block this PR. The attribute check doesn't seem that useful here given that the default is nullptr anyhow.

Reply:
Went ahead and did it.. let's see